oak_functions/client/rust/src/rekor.rs

```diff
@@ -0,0 +1,181 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//! This module provides structs for representing a Rekor LogEntry, as well as logic for parsing and
+//! verifying signatures in a Rekor LogEntry.
+
+use anyhow::Context;
+use ecdsa::Signature;
+use serde::{Deserialize, Serialize};
+use signature::Verifier;
+use std::str::FromStr;
+
+/// Struct representing a Rekor LogEntry.
+#[derive(Debug, Deserialize, PartialEq, Serialize)]
+pub struct LogEntry {
+    #[serde(rename = "body")]
+    pub body: String,
+
+    #[serde(rename = "integratedTime")]
+    pub integrated_time: usize,
+
+    // TODO(#2316): should this be verified?
+    /// This is the SHA256 hash of the DER-encoded public key for the log at the time the entry was
+    /// included in the log
+    /// Pattern: ^[0-9a-fA-F]{64}$
+    #[serde(rename = "logID")]
+    pub log_id: String,
+
+    /// Minimum: 0
+    #[serde(rename = "logIndex")]
+    pub log_index: usize,
```

Comment:
Is this really a usize? Is there a chance that it could grow too big for 32 bit clients?